services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -145,14 +145,31 @@ def mock_port_keys() -> List[str]:
     return ["first_port", "second_port"]
 
 
+@pytest.fixture
+def mutable_settings(test_client: TestClient) -> DynamicSidecarSettings:
+    settings: DynamicSidecarSettings = test_client.application.state.settings
+    # disable mutability for this test
+    settings.__config__.allow_mutation = True
+    settings.__config__.frozen = False
+    return settings
```

Comment:
Even if I use the mocking library, I cannot change the value since the settings are immutable.

Reply:
as a matter of fact you can.
and also using the monkeypatch works.
For your problem with having both inside the docker container (because you pass the host to the sidecar) and in the host, just use the host real IP. since the ports are open in testing this will work